modules/sdk-core/test/unit/account-lib/mpc/tss/ecdsa/ecdsa.ts

```diff
@@ -36,11 +36,33 @@ describe('ecdsa tss', function () {
       ecdsa.keyShare(3, 2, 3),
     ]);
 
-    const [keyCombined1, keyCombined2] = [
+    const [keyCombined1, keyCombined2, keyCombined3] = [
       ecdsa.keyCombine(keyShare1.pShare, [keyShare2.nShares[1], keyShare3.nShares[1]]),
       ecdsa.keyCombine(keyShare2.pShare, [keyShare1.nShares[2], keyShare3.nShares[2]]),
+      ecdsa.keyCombine(keyShare3.pShare, [keyShare1.nShares[3], keyShare2.nShares[3]]),
     ];
 
+    keyCombined1.xShare.y.should.equal(keyCombined2.xShare.y);
+    keyCombined1.xShare.y.should.equal(keyCombined3.xShare.y);
+
+    keyCombined1.xShare.schnorrProofX?.should.not.be.undefined();
+    keyCombined2.xShare.schnorrProofX?.should.not.be.undefined();
+    keyCombined3.xShare.schnorrProofX?.should.not.be.undefined();
+
+    // Collect all VSS from nShares and verify Schnorr proofs against X_i.
+    // Note that this is something WP needs to do after keyCombine of all participants.
```

Comment:
A party will produce the proof when it runs keyCombine or keyDerive, but it likely doesn't have all the public VSSs to verify it - it usually only gets VSSs for its index from other parties.

So to verify one such proof from any party, an intermediary will need to collect all the VSSs (and the public key).

In our case, keyCombine/keyDerive could be run at the beginning of signing instead of key gen, so if only 2 parties participate, then only proofs from those 2 need to be verified.